src/java.base/share/classes/java/lang/foreign/Arena.java

```diff
@@ -70,34 +135,120 @@
+ * like a confined arena (i.e., single-threaded access), but internally uses a
+ * {@linkplain SegmentAllocator#slicingAllocator(MemorySegment) slicing allocator} to respond to allocation requests.
+ * When the slicing arena is closed, the underlying confined arena is also closed; this will invalidate all segments
+ * allocated with the slicing arena (since the scope of the slicing arena is the same as that of the underlying
+ * confined arena):
+ *
+ * {@snippet lang = java:
+ * class SlicingArena implements Arena {
+ *      final Arena arena = Arena.ofConfined();
+ *      final SegmentAllocator slicingAllocator;
+ *
+ *      SlicingArena(long size) {
+ *          slicingAllocator = SegmentAllocator.slicingAllocator(arena.allocate(size));
+ *      }
+ *
+ *      public void allocate(long byteSize, long byteAlignment) {
+ *          return slicingAllocator.allocate(byteSize, byteAlignment);
+ *      }
+ *
+ *      public MemorySegment.Scope scope() {
+ *          return arena.scope();
+ *      }
+ *
+ *      public void close() {
+ *          return arena.close();
+ *      }
+ * }
+ * }
+ *
+ * In other words, a slicing arena provides a vastly more efficient and scalable allocation strategy, while still retaining
+ * the timely deallocation guarantee provided by the underlying confined arena:
+ *
+ * {@snippet lang = java:
+ * try (Arena slicingArena = new SlicingArena(1000)) {
+ *      for (int i = 0 ; i < 10 ; i++) {
+ *          MemorySegment s = slicingArena.allocateArray(JAVA_INT, 1, 2, 3, 4, 5);
+ *          ...
+ *      }
+ * } // all memory allocated is released here
+ * }
+ *
+ * @implSpec
+ * Implementations of this interface are thread-safe.
+ *
+ * @see MemorySegment
  *
  * @since 20
  */
 @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
 public interface Arena extends SegmentAllocator, AutoCloseable {
 
+    /**
+     * Creates a new arena that is managed, automatically, by the garbage collector.
+     * Segments obtained with the returned arena can be
+     * {@linkplain MemorySegment#isAccessibleBy(Thread) accessed} by any thread.
+     * Calling {@link #close()} on the returned arena will result in an {@link UnsupportedOperationException}.
+     *
+     * @return a new arena that is managed, automatically, by the garbage collector.
+     */
+    static Arena ofAuto() {
+        return MemorySessionImpl.createImplicit(CleanerFactory.cleaner()).asArena();
+    }
+
+    /**
+     * Obtains the global arena. Segments obtained with the global arena can be
+     * {@linkplain MemorySegment#isAccessibleBy(Thread) accessed} by any thread.
+     * Calling {@link #close()} on the returned arena will result in an {@link UnsupportedOperationException}.
+     *
+     * @return the global arena.
+     */
+    static Arena global() {
+        class Holder {
+            final static Arena GLOBAL = MemorySessionImpl.GLOBAL.asArena();
```

Comment:
Nit: use blessed modifiers order
```suggestion
            static final Arena GLOBAL = MemorySessionImpl.GLOBAL.asArena();
```